### PR TITLE
passerine: 0.9.2 → 0.9.3

### PR DESCRIPTION
**Pull request by Kozova1**
Signed-off-by: Kozova1 <mug66kk@gmail.com>

###### Motivation for this change
Passerine has released a new version and hasn't been updated yet for a while.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [X] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/passerine/default.nix [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
   pname = "passerine";
-  version = "0.9.2";
+  version = "0.9.3";
 
   src = fetchFromGitHub {
     owner = "vrtbl";
     repo = "passerine";
-    rev = "dd8a6f5efc5dcb03d45b102f61cc8a50d46e8e98";
-    sha256 = "sha256-/QzqKLkxAVqvTY4Uft1qk7nJat6nozykB/4X1YGqu/I=";
+    rev = "30757a528c1656c15ed3403d355d2b9dee028991";
```
Comment (bobby285271) [inline pkgs/development/compilers/passerine/default.nix:10]: Maybe we can reuse version again?

```suggestion
    rev = "v${version}";
```

Comment (Kozova1) [reply pkgs/development/compilers/passerine/default.nix:10]: Hi, I don't really have the time to check whether this breaks something - have you tested this?

Comment (bobby285271) [reply pkgs/development/compilers/passerine/default.nix:10]: sha256 should remain the same so I don't think this will break something...

**Review by c0bw3b — APPROVED** [review]

File: pkgs/development/compilers/passerine/default.nix [hunk 2]
```diff
@@ -2,16 +2,16 @@
...
 rustPlatform.buildRustPackage rec {
   pname = "passerine";
-  version = "0.9.2";
+  version = "0.9.3";
 
   src = fetchFromGitHub {
     owner = "vrtbl";
     repo = "passerine";
-    rev = "dd8a6f5efc5dcb03d45b102f61cc8a50d46e8e98";
-    sha256 = "sha256-/QzqKLkxAVqvTY4Uft1qk7nJat6nozykB/4X1YGqu/I=";
```
Comment (Kozova1) [inline pkgs/development/compilers/passerine/default.nix:11]: btw, what is this style of sha256? I never saw it written like this before

Comment (c0bw3b) [reply pkgs/development/compilers/passerine/default.nix:11]: It is a [Subresource Integrity (SRI)](https://www.w3.org/TR/SRI/) hash, where the hashing algo is part of the hash representation. The second part is the actual hash in base64.

SRI hashes are supported by many fetchers in the nix ecosystem now. But not all! `fetchpatch` is a notable exception at this point in time.

**Pull request merged**